wallet/wallet.go

```diff
@@ -301,6 +302,34 @@ func (w *SingleAddressWallet) Redistribute(cs consensus.State, outputs int, amou
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
+	// build map of inputs currently in the tx pool
+	inPool := make(map[types.Hash256]bool)
+	for _, ptxn := range pool {
+		for _, in := range ptxn.SiacoinInputs {
+			inPool[types.Hash256(in.ParentID)] = true
+		}
+	}
+
+	// fetch unspent transaction outputs
+	utxos, err := w.store.UnspentSiacoinElements(false)
+	if err != nil {
+		return types.Transaction{}, nil, err
+	}
+
+	// check whether a redistribution is necessary, adjust number of desired
+	// outputs accordingly
+	for _, sce := range utxos {
+		inUse := w.isOutputUsed(sce.ID) || inPool[sce.ID]
+		matured := cs.Index.Height >= sce.MaturityHeight
```

Comment:
I think you can drop this line if you call `UnspentSiacoinElements(true` instead of calling it with `false`.

Reply:
I thought about that but the reason I kept it is because I did not want to lose the extra info in the logs:

```
inputs %v < needed %v + txnFee %v (usable: %v, inUse: %v, sameValue: %v, notMatured: %v)",
```

If figured it's nice to have this info if wallet maintenance fails.